### PR TITLE
Loading spinner

### PR DESCRIPTION
**Pull request by falko17**
This implements a loading spinner, intended to be displayed for long-running operations.
The spinner is displayed in the upper left corner of the screen (as the other corners are already occupied) – if there is more than one long-running process active, a small number will be displayed indicating how many processes are currently running. Hovering over the spinner yields a tooltip containing a short message describing the running processes.

## Usage
The loading spinner can be accessed via static methods. A short, and especially a **unique** message should be passed to the `Show` method – this is what's shown to the user.
This message is used to identify the loading process, and is later used to hide the spinner again once the process has completed running. If no message is passed to `Show`, a random one will be generated to ensure uniqueness.

### Recommended method
```csharp
using (LoadingSpinner.Show("Loading message")) 
{
     // Do something that takes a long time here.
     // The loading spinner will be shown while this is running,
     // and hidden once control flow leaves this block.
}
```

### Alternative: Using the `IDisposable`
```csharp
IDisposable disposable = LoadingSpinner.Show("Loading message"); 
// Do something that takes a long time here.
disposable.Dispose();
```

### Alternative: Using the message
```csharp
LoadingSpinner.Show("Loading message");
// Do something that takes a long time here.
LoadingSpinner.Hide("Loading message");  // must be the same message!
```

## Additional changes
* A utility method `RunOnMainThread` has been implemented by request of @Leon2796. When called from the thread pool, it switches to the main thread only for the given action. This is useful because certain methods and properties are not accessible from the thread pool.
* A bug in the tooltip has been fixed that lead to it appearing when it is disabled and then enabled.

### CODE REVIEW

File: Assets/SEE/UI/LoadingSpinner.cs
Comment (falko17) [inline Assets/SEE/UI/LoadingSpinner.cs:1]: @koschke
Currently, the loading spinner is shown during these processes:
* A `SEECity` is loading
* The `EdgeMeshScheduler` is converting line-rendered edges into meshes

Are there any other cases for which we want to display the loading spinner?

File: Assets/SEE/UI/LoadingSpinner.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        ///
+        /// Refer to the class documentation for more information on how to use this.
+        /// </summary>
+        /// <param name="processMessage">The unique message for the loading process</param>
+        /// <seealso cref="LoadingSpinner"/>
+        /// <returns>An <see cref="IDisposable"/> that can be used to hide the spinner.</returns>
+        public static IDisposable Show(string processMessage = null)
+        {
+            // If the process name is not specified, we generate a random one.
+            // TODO (@koschke): However, does it ever make sense to not specify a process name?
```
[the remark this answers is not in the document]
Comment (koschke) [reply Assets/SEE/UI/LoadingSpinner.cs:99]: I can't image a situation in which specifying a process message would not be doable. I think it makes sense to require it.
If we ever run into a situation where insisting on a process message were a real blocker, we could loosen this requirement.

File: Assets/SEE/UI/LoadingSpinner.cs
Comment (koschke) [reply Assets/SEE/UI/LoadingSpinner.cs:1]: Loading and drawing a graph can be expensive operations, too. We can do these a runtime.

Comment (koschke) [reply Assets/SEE/UI/LoadingSpinner.cs:1]: What about the evolution rendering (drawing a series of graphs)? That is even more expensive.

File: Assets/SEE/Utils/AsyncUtils.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        public static async UniTask<T> RunOnMainThread<T>(Func<T> action)
+        {
+            bool notOnMainThread = Thread.CurrentThread.ManagedThreadId != 1;
+            if (notOnMainThread)
+            {
+                await UniTask.SwitchToMainThread();
+            }
+            else
+            {
+                Debug.LogWarning("RunOnMainThread called from main thread. This is not necessary.");
```
Comment (koschke) [inline Assets/SEE/Utils/AsyncUtils.cs:39]: A `\n` should be added to the end of the string literal.

**Review by koschke — CHANGES_REQUESTED** [review]
Looks good to me. Only one minor change request. 

I answered the TODO questions. You can replace the TODO comments accordingly.

File: Assets/SEE/UI/LoadingSpinner.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (falko17) [reply Assets/SEE/UI/LoadingSpinner.cs:99]: Alright, I've changed the parameter to be required.

File: Assets/SEE/Utils/AsyncUtils.cs (hunk 2 above)
Comment (falko17) [reply Assets/SEE/Utils/AsyncUtils.cs:39]: Fixed it.

File: Assets/SEE/UI/LoadingSpinner.cs
Comment (falko17) [reply Assets/SEE/UI/LoadingSpinner.cs:1]: Good idea, the loading spinner is now shown in both of those cases.

**Review by koschke — APPROVED** [review]

**Pull request merged**